test/op.cpp

```diff
@@ -41,3 +43,35 @@ TEST(op, is_parameter)
     ASSERT_NE(nullptr, t0);
     EXPECT_FALSE(t0->is_parameter());
 }
+
+TEST(op, provenance_tag)
+{
+    auto node = make_shared<op::Parameter>(element::f32, Shape{1});
+    auto tag1 = "parameter node";
+    auto tag2 = "f32 node";
+    node->add_provenance_tag(tag1);
+    node->add_provenance_tag(tag2);
+
+    node->remove_provenance_tag(tag1);
+
+    auto tags = node->get_provenance_tags();
+    ASSERT_TRUE(tags.find(tag1) == tags.end());
+    ASSERT_TRUE(tags.find(tag2) != tags.end());
+}
+
+// TODO: Need to mock Node, Op etc to be able to unit test functions like replace_node().
```

Comment:
(We've already chatted a bit about this on Slack, just summarizing my understanding on the PR thread.) Let's work without mocks for now, until we have a chance to look into applying them more broadly.

Reply:
In my updated push, I left my commented code for future reminder if that's ok.